backend/infrahub/core/branch/tasks.py

```diff
@@ -412,3 +413,49 @@ async def _get_diff_root(
         )
 
     return default_branch_diff
+
+
+@flow(
+    name="branch-merge-post-process",
+    flow_run_name="Run additional tasks after merging {model.source_branch} in {model.target_branch}",
+)
+async def post_process_branch_merge(model: BranchMergePostProcessModel, service: InfrahubServices) -> None:
+    async with service.database.start_session() as db:
+        await add_tags(branches=[model.source_branch])
+        log = get_run_logger()
+        log.info(f"Running additional tasks after merging {model.source_branch} within {model.target_branch}")
+
+        events: list[InfrahubMessage] = [
+            messages.RefreshRegistryBranches(),
+        ]
+        component_registry = get_component_registry()
+        default_branch = registry.get_branch_from_registry()
+        diff_repository = await component_registry.get_component(DiffRepository, db=db, branch=default_branch)
+        # send diff update requests for every branch-tracking diff
+        branch_diff_roots = await diff_repository.get_roots_metadata(base_branch_names=[model.target_branch])
+
+        await service.workflow.submit_workflow(
```

Comment:
I think we should move  `TRIGGER_ARTIFACT_DEFINITION_GENERATE` and `TRIGGER_GENERATOR_DEFINITION_RUN` to a trigger instead of having it in a flow

Reply:
I think we said for a few of these migrations that there were improvements that could be done overall but that that the initial goal should probably be to just get them over to Prefect in the first place.